src/main/java/au/com/titanclass/flatmate/FlatmateContext.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2018 Titan Class Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package au.com.titanclass.flatmate;
+
+import javax.naming.*;
+import java.util.Hashtable;
+
+public class FlatmateContext implements Context {
+
+  private static Hashtable<String, Object> things = new Hashtable<>();
```

Comment:
This will be required to a `ConcurrentHashMap` or at least synchronised as per the info at: https://docs.oracle.com/javase/7/docs/api/javax/naming/Context.html

Reply:
Changed to ConcurrentHashMap (Hashtable is thread safe but ConcurrentHashMap is recommended from a performance perspective)